libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -114,15 +114,15 @@ export class GRPCService implements Service {
     context: EvaluationContext,
     logger: Logger
   ): Promise<ResolutionDetails<boolean>> {
-    return this.resolve(this._client.resolveBoolean, flagKey, context, logger);
+    return this.resolve(this._client.resolveBoolean, flagKey, context, logger, this.booleanParser);
   }
 
   async resolveString(flagKey: string, context: EvaluationContext, logger: Logger): Promise<ResolutionDetails<string>> {
-    return this.resolve(this._client.resolveString, flagKey, context, logger);
+    return this.resolve(this._client.resolveString, flagKey, context, logger, this.stringParser);
   }
 
   async resolveNumber(flagKey: string, context: EvaluationContext, logger: Logger): Promise<ResolutionDetails<number>> {
-    return this.resolve(this._client.resolveFloat, flagKey, context, logger);
+    return this.resolve(this._client.resolveFloat, flagKey, context, logger, this.numberParser);
```

Comment:
So it looks like this wasn't working as expected? When a zero-value flag was set in flagd config, `undefined` was returned from the provider? Did you see that in a manual test?

Reply:
yes exactly, this logic was not in place

Reply:
Perfect! This was a productive issue then.

Thanks @odubajDT and @james-milligan